### ISSUE TITLE
initContainers should not request CPU

### ISSUE THREAD

**Issue by sebbonnet** [sky-uk/cassandra-operator#213]
### What is the bug being reported?
The definition of the statefulset for a Cassandra rack requests the same amount of CPU for each initContainer as is requested for the main Cassandra container. This can cause scheduling problems on Kubernetes clusters which use the CPU Management feature.

### What would be the expected behavior?
initContainers should not request any CPU.

### If repeatable, how can we reproduce this issue?
Create any Cassandra cluster with the operator and look at the initContainer definitions on the statefulsets. You'll see that the initContainers are requesting the same amount of CPU as is requested for the main container in the Cassandra spec.
This may result in the init container start up to fail as it attemps to request additional CPU on top of the ones already allocated to the main cassandra container - e.g. `Internal PreStartContainer hook failed: not enough cpus available to satisfy request`

### Workaround, if known
Scale down the operator and manually modify the statefulsets.
 
### Your Environment
Include as many relevant details as possible about the environment you experienced the bug in:
* Kubernetes version: 1.12
* Cassandra operator docker image: v0.77.0-alpha
* Cassandra cluster definition - e.g. `kubectl get cassandra my_cluster -o yaml`: 
```yaml
apiVersion: core.sky.uk/v1alpha1
kind: Cassandra
metadata:                                              
  name: my-name
  namespace: my-namespace
spec:
  datacenter: test
  pod:
    image: cassandra:3.11
    resources:
      limits:
        cpu: "4"
        memory: 2Gi
      requests:
        cpu: "4"
        memory: 2Gi
  racks:
  - name: a
    replicas: 1
    storage:
    - persistentVolumeClaim:
        resources:
          requests:
            storage: 10Gi
        storageClassName: standard-zone-a
    zone: eu-west-1a
  - name: b
    replicas: 1
    storage:
    - persistentVolumeClaim:
        resources:
          requests:
            storage: 10Gi
        storageClassName: standard-zone-b
    zone: eu-west-1b

```
* Statefulset definition:
```yaml
apiVersion: apps/v1
kind: StatefulSet
metadata:
  [ ... ]
spec:
  podManagementPolicy: OrderedReady
  replicas: 1
  revisionHistoryLimit: 10
  selector:
    matchLabels:
      app: my-cassandra
      rack: a
      sky.uk/cassandra-operator: my-cassandra
  serviceName: my-cassandra
  template:
    metadata:
      annotations:
        clusterConfigHash: 4c6e4209d86179c28b4fab775d0e52dcb55fc00acaf70a3652bb44b38af83daf
      creationTimestamp: null
      labels:
        app: my-cassandra
        rack: a
        sky.uk/cassandra-operator: my-cassandra
    spec:
      affinity:
        nodeAffinity:
          requiredDuringSchedulingIgnoredDuringExecution:
            nodeSelectorTerms:
            - matchExpressions:
              - key: failure-domain.beta.kubernetes.io/zone
                operator: In
                values:
                - eu-west-1a
        podAntiAffinity:
          requiredDuringSchedulingIgnoredDuringExecution:
          - labelSelector:
              matchLabels:
                sky.uk/cassandra-operator: my-cassandra
            topologyKey: kubernetes.io/hostname
      containers:
      - env:
        - name: EXTRA_CLASSPATH
          value: /extra-lib/cassandra-seed-provider.jar
        image: cassandra:3.11
        imagePullPolicy: IfNotPresent
        lifecycle:
          preStop:
            exec:
              command:
              - /bin/sh
              - -c
              - nodetool drain
        livenessProbe:
          failureThreshold: 3
          httpGet:
            path: /live
            port: 8080
            scheme: HTTP
          initialDelaySeconds: 30
          periodSeconds: 30
          successThreshold: 1
          timeoutSeconds: 5
        name: cassandra
        ports:
        - containerPort: 7000
          name: internode
          protocol: TCP
        - containerPort: 7070
          name: jmx-exporter
          protocol: TCP
        - containerPort: 7199
          name: cassandra-jmx
          protocol: TCP
        - containerPort: 7777
          name: jolokia
          protocol: TCP
        - containerPort: 9042
          name: client
          protocol: TCP
        readinessProbe:
          failureThreshold: 3
          httpGet:
            path: /ready
            port: 8080
            scheme: HTTP
          initialDelaySeconds: 30
          periodSeconds: 15
          successThreshold: 1
          timeoutSeconds: 5
        resources:
          limits:
            cpu: "4"
            memory: 2Gi
          requests:
            cpu: "4"
            memory: 2Gi
        terminationMessagePath: /dev/termination-log
        terminationMessagePolicy: File
        volumeMounts:
        - mountPath: /etc/cassandra
          name: configuration
        - mountPath: /extra-lib
          name: extra-lib
        - mountPath: /var/lib/cassandra
          name: var-lib-cassandra
      - args:
        - --log-level=info
        - --health-server-port=8080
        env:
        - name: CLUSTER_NAMESPACE
          value: my-namespace
        - name: CLUSTER_NAME
          value: my-cassandra
        - name: CLUSTER_CURRENT_RACK
          value: a
        - name: CLUSTER_DATA_CENTER
          value: test
        - name: NODE_LISTEN_ADDRESS
          valueFrom:
            fieldRef:
              apiVersion: v1
              fieldPath: status.podIP
        - name: POD_CPU_MILLICORES
          value: "4000"
        - name: POD_MEMORY_BYTES
          value: "2147483648"
        image: skyuk/cassandra-sidecar:v0.78.0-alpha
        imagePullPolicy: IfNotPresent
        name: cassandra-sidecar
        ports:
        - containerPort: 8080
          name: api
          protocol: TCP
        resources:
          limits:
            cpu: 100m
            memory: 50Mi
          requests:
            cpu: 100m
            memory: 50Mi
        terminationMessagePath: /dev/termination-log
        terminationMessagePolicy: File
      dnsPolicy: ClusterFirst
      initContainers:
      - command:
        - sh
        - -c
        - cp -vr /etc/cassandra/* /configuration
        image: cassandra:3.11
        imagePullPolicy: IfNotPresent
        name: init-config
        resources:
          limits:
            cpu: "4"
            memory: 2Gi
          requests:
            cpu: "4"
            memory: 2Gi
        terminationMessagePath: /dev/termination-log
        terminationMessagePolicy: File
        volumeMounts:
        - mountPath: /configuration
          name: configuration
      - env:
        - name: CLUSTER_NAMESPACE
          value: my-namespace
        - name: CLUSTER_NAME
          value: my-cassandra
        - name: CLUSTER_CURRENT_RACK
          value: a
        - name: CLUSTER_DATA_CENTER
          value: test
        - name: NODE_LISTEN_ADDRESS
          valueFrom:
            fieldRef:
              apiVersion: v1
              fieldPath: status.podIP
        - name: POD_CPU_MILLICORES
          value: "4000"
        - name: POD_MEMORY_BYTES
          value: "2147483648"
        image: skyuk/cassandra-bootstrapper:v0.78.0-alpha
        imagePullPolicy: IfNotPresent
        name: cassandra-bootstrapper
        resources:
          limits:
            cpu: "4"
            memory: 2Gi
          requests:
            cpu: "4"
            memory: 2Gi
        terminationMessagePath: /dev/termination-log
        terminationMessagePolicy: File
        volumeMounts:
        - mountPath: /configuration
          name: configuration
        - mountPath: /extra-lib
          name: extra-lib
        - mountPath: /custom-config
          name: cassandra-custom-config-my-cassandra
      restartPolicy: Always
      schedulerName: default-scheduler
      securityContext:
        fsGroup: 999
        runAsUser: 999
      serviceAccount: cassandra-node
      serviceAccountName: cassandra-node
      terminationGracePeriodSeconds: 30
      volumes:
      - emptyDir: {}
        name: configuration
      - emptyDir: {}
        name: extra-lib
      - configMap:
          defaultMode: 420
          name: my-cassandra-config
        name: cassandra-custom-config-my-cassandra
  updateStrategy:
    rollingUpdate:
      partition: 0
    type: RollingUpdate
  volumeClaimTemplates:
  - metadata:
      creationTimestamp: null
      labels:
        app: my-cassandra
        rack: a
        sky.uk/cassandra-operator: my-cassandra
      name: var-lib-cassandra
      namespace: my-namespace
    spec:
      accessModes:
      - ReadWriteOnce
      dataSource: null
      resources:
        requests:
          storage: 10Gi
      storageClassName: standard-zone-a
    status:
      phase: Pending
```

**Comment by sebbonnet** [comment]
We should also confirm whether assigning resource quotas to the cassandra-sidecar affects the cpu affinity.

**Comment by sebbonnet** [comment]
Closed by #215
